libs/watt/package/menu/+storybook/menu-async-story.component.ts

```diff
@@ -0,0 +1,122 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+//#endregion
+import { Component, input, signal, effect, OnDestroy } from '@angular/core';
+import { WattButtonComponent } from '../../button';
+import { WattIconComponent } from '../../icon';
+import { WattMenuComponent } from '../watt-menu.component';
+import { WattMenuItemComponent } from '../watt-menu-item.component';
+import { WattMenuTriggerDirective } from '../watt-menu-trigger.directive';
+
+@Component({
+  selector: 'watt-menu-async-story',
+  imports: [
+    WattButtonComponent,
+    WattIconComponent,
+    WattMenuComponent,
+    WattMenuItemComponent,
+    WattMenuTriggerDirective,
+  ],
+  template: `
+    <div style="display: flex; gap: 20px; flex-wrap: wrap;">
+      <div>
+        <h3>Async Menu Items (Dynamic Icons)</h3>
+        <watt-button
+          variant="secondary"
+          [wattMenuTriggerFor]="asyncMenu"
+          aria-label="Open async menu"
+        >
+          Async Dynamic Icons
+          <watt-icon name="down" size="xs" />
+        </watt-button>
+
+        <watt-menu #asyncMenu>
+          @if (itemsLoaded()) {
+            <watt-menu-item (click)="onAction('Profile')">
+              @if (showIcons()) {
+                <watt-icon name="user" />
+              }
+              Profile
+            </watt-menu-item>
+            <watt-menu-item (click)="onAction('Settings')">
+              Settings (always no icon)
+            </watt-menu-item>
+            <watt-menu-item (click)="onAction('Preferences')">
+              @if (showIcons()) {
+                <watt-icon name="settings" />
+              }
+              Preferences
+            </watt-menu-item>
+            <watt-menu-item (click)="onAction('Help')"> Help (always no icon) </watt-menu-item>
+            <watt-menu-item (click)="onAction('Logout')">
+              @if (showIcons()) {
+                <watt-icon name="logout" />
+              }
+              Logout
+            </watt-menu-item>
+          }
+        </watt-menu>
+      </div>
+    </div>
+
+    <p style="margin-top: 20px">Last action: {{ lastAction() || 'None' }}</p>
+    <p>Items loaded: {{ itemsLoaded() ? 'Yes' : 'No (loading...)' }}</p>
+  `,
+})
+export class MenuAsyncStoryComponent implements OnDestroy {
+  showIcons = input(true);
+  loadDelay = input(1000);
+
+  private loadingTimeout?: ReturnType<typeof setTimeout>;
+
+  itemsLoaded = signal(false);
+  lastAction = signal<string | null>(null);
+
+  constructor() {
+    // Watch for loadDelay changes and restart loading
+    effect(() => {
+      this.loadDelay();
+      this.startLoading();
+    });
```

Comment:
just a little nitpick, you can use the `onCleanup` argument in `effect` to clear the timeout, then you don't need the `ngOnDestroy` and the `if (this.loadingTimeout)` check